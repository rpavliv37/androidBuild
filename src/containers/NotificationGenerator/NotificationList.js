import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { deleteNotification } from './actions';
import Notification from './Notification';
import { Block } from 'galio-framework';

/**
 * Renders list of notifications
 * @memberof module:Notification
 */
const NotificationList = ({ messages, deleteNotification: deleteNotificationFromProps }) => {
  const notifications = messages.map((message) => (
    <Notification key={message.id} message={message} deleteNotification={deleteNotificationFromProps} />
  ));
  return (
    <Block style={{borderWidth:1,position:'absolute',bottom:0,alignSelf:'flex-end'}}>
      {notifications}
    </Block>
  );
};

NotificationList.propTypes = {
  messages: PropTypes.arrayOf(PropTypes.object).isRequired,
  deleteNotification: PropTypes.func.isRequired
};

function mapStateToProps(state) {
  return {
    messages: state.notifiactions
  };
}

export default connect(mapStateToProps, {
  deleteNotification
})(NotificationList);
