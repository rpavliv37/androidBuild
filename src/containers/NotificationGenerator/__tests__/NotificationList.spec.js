import React from 'react';
import { render } from 'enzyme';
import configureStore from 'redux-mock-store';
import NotificationList from '../NotificationList';

const initialState = {
  notifiactions: [
    {
      id: 'messageID1',
      type: 'success',
      text: 'Test message. Success.'
    },
    {
      id: 'messageID2',
      type: 'error',
      text: 'Test message. Error.'
    }
  ]
};
const mockStore = configureStore();
const store = mockStore(initialState);

describe('NotificationGenerator: NotificationList', () => {
  it('Shallow render of NotificationList', () => {
    const props = {
      deleteNotification: () => {}
    };
    const wrapper = render(<NotificationList {...props} store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
