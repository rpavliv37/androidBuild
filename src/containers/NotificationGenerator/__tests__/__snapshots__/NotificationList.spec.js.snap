// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationGenerator: NotificationList Shallow render of NotificationList 1`] = `
<div
  class="toast-bottom-right"
  id="toast-container"
>
  <div
    aria-live="polite"
    class="toast toast-success"
    role="status"
  >
    <div
      class="toast-message"
    >
      Test message. Success.
    </div>
  </div>
  <div
    aria-live="polite"
    class="toast toast-error"
    role="status"
  >
    <div
      class="toast-message"
    >
      Test message. Error.
    </div>
  </div>
</div>
`;
